openlibrary/macros/BestbookModal.html

```diff
@@ -0,0 +1,79 @@
+$def with (work, link_markup, id, classes='', reload_id=None, edition_key=None)
+$# :param Work work: The work that is being reviewed
+$# :param str link_markup: arkup for element that triggers observations modal
+$# :param str id: Unique identifier for this modal
+$# :param str classes: HTML classes for this component
+$# :param str reload_id: Reference to UI component that will update when observations are added or removed
+
+$ topics = work.get_subjects()
+$ username = ctx.user and ctx.user.key.split('/')[-1]
+
+$# The following data is accessed via JS:
+$ context = {
+  $ "username": username,
+  $ "work": work.key,
+  $ "id": id
+  $ }
+
+$# Render the subject input field
+$jsdef render_subject_field(name, data):
+    <div class="input">
+        $code:
+            subjects = []
+            for s in data:
+                subjects.append('"' + s + '"' if ',' in s else s)
+            subject_str = ', '.join(subjects)
+            rows = len(subject_str) // 95 + 1
+        <textarea id="about-$name" name="work--$name" rows="$rows" wrap="soft" tabindex="0">$subject_str</textarea>
+    </div>
+
+$jsdef render_subject_autocomplete_item(item):
+    <div class="ac_subject" title="$_('Select this subject')">
+        <span class="name">$item.name</span>
+    </div>
+
+$if reload_id:
+  $ context['reloadId'] = reload_id
+
+$ work_id = work.key.split('/')[2]
+$ cover_url = "https://covers.openlibrary.org/b/id/{}-M.jpg".format(work_id[2:-1])
```

Comment:
Here the covers server works in three ways -- based on the edition ID, the work ID, or the cover ID.

The cover ID (i.e. /b/id) is a universal ID across all covers, e.g. https://covers.openlibrary.org/b/id/8000002-M.jpg which is **not** the cover for /books/OL8000002.

If given a work ID, we can use the format:
https://covers.openlibrary.org/w/olid/OL262421W-M.jpg

Notice the `/w/olid` as opposed to `/b/id` (cover ID) or `/b/olid`
(edition ID).

Given the edition, we'd use https://covers.openlibrary.org/b/olid/OL39802909M-M.jpg
(notice the `/b/olid`)

Not every work will have a cover... Where possible, if we have the ID of the edition we're on, we'll want to use the edition ID (unless there is no cover, in which case we can fallback to the work id)